src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -3033,6 +3052,326 @@ - (void)test032_MTRPathClassesEncoding
+            XCTAssertNotNil(attributePath);
+
+            if (attributePath.cluster.unsignedLongValue == MTRClusterIDTypeDescriptorID) {
+                NSDictionary * data = attributeDict[MTRDataKey];
+                XCTAssertNotNil(data);
+                switch (attributePath.attribute.unsignedLongValue) {
+                case MTRAttributeIDTypeClusterDescriptorAttributePartsListID: {
+                    dataVersionForPartsList = data[MTRDataVersionKey];
+                    XCTAssertNotNil(dataVersionForPartsList);
+                    break;
+                }
+                case MTRAttributeIDTypeClusterDescriptorAttributeDeviceTypeListID: {
+                    dataVersionForDeviceTypesList = data[MTRDataVersionKey];
+                    XCTAssertNotNil(dataVersionForDeviceTypesList);
+                    break;
+                }
+                case MTRAttributeIDTypeClusterDescriptorAttributeServerListID: {
+                    dataVersionForServerList = data[MTRDataVersionKey];
+                    XCTAssertNotNil(dataVersionForServerList);
+                    break;
+                }
+                }
+            } else if (attributePath.cluster.unsignedLongValue == MTRClusterIDTypeIdentifyID) {
+                NSDictionary * data = attributeDict[MTRDataKey];
+                XCTAssertNotNil(data);
+                switch (attributePath.attribute.unsignedLongValue) {
+                case MTRAttributeIDTypeGlobalAttributeAttributeListID: {
+                    dataVersionForAttributeList = data[MTRDataVersionKey];
+                    ;
+                    XCTAssertNotNil(dataVersionForAttributeList);
+                    break;
+                }
+                case MTRAttributeIDTypeGlobalAttributeClusterRevisionID: {
+                    dataVersionForClusterRevision = data[MTRDataVersionKey];
+                    ;
+                    XCTAssertNotNil(dataVersionForClusterRevision);
+                    break;
+                }
+                }
+            } else if (attributePath.cluster.unsignedLongValue == MTRClusterIDTypeOvenCavityOperationalStateID && attributePath.attribute.unsignedLongValue == MTRAttributeIDTypeGlobalAttributeAcceptedCommandListID) {
+                NSDictionary * data = attributeDict[MTRDataKey];
+                XCTAssertNotNil(data);
+                dataVersionForAcceptedCommandList = data[MTRDataVersionKey];
+                ;
+                XCTAssertNotNil(dataVersionForAcceptedCommandList);
+            } else if (attributePath.cluster.unsignedLongValue == MTRClusterIDTypeGroupsID && attributePath.attribute.unsignedLongValue == MTRAttributeIDTypeGlobalAttributeFeatureMapID) {
+                NSDictionary * data = attributeDict[MTRDataKey];
+                XCTAssertNotNil(data);
+                dataVersionForFeatureMap = data[MTRDataVersionKey];
+                XCTAssertNotNil(dataVersionForFeatureMap);
```

Comment:
All these `XCTAssertNotNil(dataVersionForX)` should be happening in `onReportEnd`, no?  To make sure we actually got all the attributes we expected.

Reply:
done.